framework/src/service/ServiceReferenceBasePrivate.cpp

```diff
@@ -106,7 +117,7 @@ std::shared_ptr<void> ServiceReferenceBasePrivate::GetService(BundlePrivate* bun
   auto s = ExtractInterface(GetServiceInterfaceMap(bundle), interfaceId);
   if (!s)
   {
-    registration->Lock(), --registration->dependents[bundle];
+    registration->Lock(), --registration->dependents.at(bundle);
```

Comment:
If `GetServiceInterfaceMap` above returned early, the `.at(bundle)` might throw now, which would propagate into client code. I think it would be better to `find` the bundle.

Also (this problem is not new), if `GetServiceInterfaceMap` returns before incrementing the count, we might decrement it "too much".